consensus/istanbul/core/roundstate.go

```diff
@@ -177,22 +234,34 @@ func (s *roundState) CreateAndSetPreparedCertificate(quorumSize int) error {
 		return errFailedCreatePreparedCertificate
 	}
 	s.preparedCertificate = istanbul.PreparedCertificate{
-		Proposal:                s.Preprepare.Proposal,
+		Proposal:                s.preprepare.Proposal,
 		PrepareOrCommitMessages: messages,
 	}
 	return nil
 }
 
+func (s *roundStateImpl) PreparedCertificate() istanbul.PreparedCertificate {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.preparedCertificate
+}
+
+func (s *roundStateImpl) SetPreparedCertificate(preparedCertificate istanbul.PreparedCertificate) {
```

Comment:
Is this only used for testing? If so, should we remove it?

Reply:
yes